benchmarks/overrides_benchmark/common.py

```diff
@@ -14,7 +14,7 @@ def __init__(self, data, requires_grad=False):
             self._tensor = data
             return
 
-        self._tensor = torch.Tensor(data, requires_grad)
+        self._tensor = torch.empty(data, requires_grad)
```

Comment:
This doesn't look right, should be `torch.tensor`?

Reply:
There's lots of similar changes in gh-53889. What am I missing?

Reply:
AFAIK, this one in particular is called from either two places:

https://github.com/pytorch/pytorch/blob/f2b4b0e9eb670ad79f4460c0d45174f879b523f9/benchmarks/overrides_benchmark/pyspybench.py#L24-L25

https://github.com/pytorch/pytorch/blob/f2b4b0e9eb670ad79f4460c0d45174f879b523f9/benchmarks/overrides_benchmark/bench.py#L54-L55

The behavior of `torch.Tensor` with a scalar is the same as that of `torch.empty`. `torch.tensor` would make sense if `data` was something iterable.

Reply:
That said, maybe the intended behavior was `torch.Tensor([1])` and `torch.Tensor([2])`, which would make `torch.tensor` the correct one.

Reply:
I'd ensure the input is a list (or tensor/tensorlike).

The second file you link to has another use of the legacy constructor that needs to be removed:
```
    types = torch.Tensor, SubTensor, WithTorchFunction, SubWithTorchFunction

    for t in types:
        tensor_1 = t(1)
        tensor_2 = t(2)
```

Reply:
That makes sense.
By the way, what should happen with `torch.Tensor` sub-classes, such as `SubTensor` and `SubWithTorchFunction`, then? They will call the legacy constructor, right!?

Reply:
That's a great question. Yes, they will call the legacy constructor, and it'll be a little hard to avoid. 

https://pytorch.org/docs/stable/notes/extending.html explains how to write a basic subclass - with the deprecated constructor that's no longer the best option. I think the right solution is make the subclass have a `__new__` method, which uses `torch.tensor` to instantiate a tensor, and then create the subclass via taking a custom view of it (similar to what `numpy.ma.MaskedArray` does, see https://github.com/numpy/numpy/blob/main/numpy/ma/core.py#L2836). Until we sort that out, subclasses have to live with the deprecation warning.

I suggest opening a separate issue for that.